splinterd/src/daemon.rs

```diff
@@ -133,28 +132,65 @@ impl SplinterDaemon {
-        let storage_location = match &self.storage_type as &str {
-            "yaml" => Path::new(&self.state_dir)
-                .join("circuits.yaml")
-                .to_str()
-                .ok_or_else(|| {
-                    StartError::StorageError("'state_dir' is not a valid UTF-8 string".into())
-                })?
-                .to_string(),
-            "memory" => "memory".to_string(),
-            _ => {
-                return Err(StartError::StorageError(format!(
-                    "storage type is not supported: {}",
-                    self.storage_type
-                )))
+        let admin_service_store = {
+            if let Some(storage) = &self.storage_type {
+                // Load initial state from the configured storage type and state directory, then create
+                // the new SplinterState from the retrieved circuit directory
+                match &storage as &str {
+                    "yaml" => {
+                        let circuits_location = Path::new(&self.state_dir)
+                            .join("circuits.yaml")
+                            .to_str()
+                            .ok_or_else(|| {
+                                StartError::StorageError(
+                                    "'state_dir' is not a valid UTF-8 string".into(),
+                                )
+                            })?
+                            .to_string();
+                        let proposals_location = Path::new(&self.state_dir)
+                            .join("circuit_proposals.yaml")
+                            .to_str()
+                            .ok_or_else(|| {
+                                StartError::StorageError(
+                                    "'state_dir' is not a valid UTF-8 string".into(),
+                                )
+                            })?
+                            .to_string();
+
+                        Box::new(
+                            YamlAdminServiceStore::new(circuits_location, proposals_location)
+                                .map_err(|err| {
+                                    StartError::StorageError(format!(
+                                        "Unable to create YamlAdminServiceStore: {}",
+                                        err
+                                    ))
+                                })?,
+                        )
+                    }
+                    "memory" => {
+                        let store_factory = create_store_factory("memory")?;
+                        store_factory.get_admin_service_store()
+                    }
+                    _ => {
+                        return Err(StartError::StorageError(format!(
+                            "storage type is not supported: {}",
+                            storage
+                        )))
+                    }
+                }
+            } else {
+                //
```

Comment:
Looks like a comment that never got filled out

Reply:
will fix in another PR